sycl/include/sycl/properties/accessor_properties.hpp

```diff
@@ -118,12 +99,26 @@ template <typename DataT, int Dimensions, access::mode AccessMode,
           access::target AccessTarget, access::placeholder IsPlaceholder,
           typename PropertyListT>
 class accessor;
-template <typename DataT, int Dimensions, access::mode AccessMode,
-          access::target AccessTarget, access::placeholder IsPlaceholder>
-class image_accessor;
 template <typename DataT, int Dimensions, access::mode AccessMode>
 class host_accessor;
 
+namespace detail {
+namespace acc_properties {
+template <typename T> struct is_accessor : std::false_type {};
+template <typename DataT, int Dimensions, access::mode AccessMode,
+          access::target AccessTarget, access::placeholder IsPlaceholder,
+          typename PropertyListT>
+struct is_accessor<accessor<DataT, Dimensions, AccessMode, AccessTarget,
+                            IsPlaceholder, PropertyListT>> : std::true_type {};
+
+
+template <typename T> struct is_host_accessor : std::false_type {};
+template <typename DataT, int Dimensions, access::mode AccessMode>
+struct is_host_accessor<host_accessor<DataT, Dimensions, AccessMode>>
+    : std::true_type {};
+} // namespace acc_properties
+} // namespace detail
```

Comment:
Alternatively, we can move that into `namespace detail` inside the `accessor.hpp`.

Reply:
I don't have a preference. If it becomes relevant for other cases, it should be simple to move them then.